test/unittest/rtps/persistence/PersistenceGuidTests.cpp

```diff
@@ -0,0 +1,435 @@
+    {
+        return nullptr;
+    }
+
+    void deleteData(
+            void* /*data*/) override
+    {
+    }
+
+    bool getKey(
+            void* /*data*/,
+            InstanceHandle_t* /*ihandle*/,
+            bool /*force_md5*/) override
+    {
+        return true;
+    }
+
+};
+
+
+/*!
+ * @fn TEST(PersistenceGuidTest, DDS_persistence_guid)
+ * @brief This test checks if the persistence guid is set correctly on the RTPSWriter when it is specified using the property
+ * dds.persistence.guid through DDS layer
+ */
+TEST(PersistenceGuidTest, DDS_persistence_guid)
+{
+    HelloWorld hello;
+    hello.message("HelloWorld");
+
+    // Create participant
+    DomainParticipantQos pqos;
+    pqos.properties().properties().emplace_back("dds.persistence.plugin", "builtin.SQLITE3");
+    pqos.properties().properties().emplace_back("dds.persistence.sqlite3.filename", "persistence.db");
+    DomainParticipant* participant = DomainParticipantFactory::get_instance()->create_participant(0, pqos);
+    ASSERT_NE(participant, nullptr);
+
+    /* Register participant on type */
+    TypeSupport type(new TopicDataTypeMock());
+    type.register_type(participant);
+
+    // Create a publisher and a subscriber
+    Publisher* publisher = participant->create_publisher(PUBLISHER_QOS_DEFAULT, nullptr);
+    ASSERT_NE(publisher, nullptr);
+    Subscriber* subscriber = participant->create_subscriber(SUBSCRIBER_QOS_DEFAULT, nullptr);
+    ASSERT_NE(subscriber, nullptr);
+
+    // Create a topic
+    Topic* topic = participant->create_topic("persistence_topic_name", type.get_type_name(), TOPIC_QOS_DEFAULT);
+    ASSERT_NE(topic, nullptr);
+
+    // Configure DataWriter's durability and persistence GUID so it can use the persistence service
+    DataWriterQos dwqos = DATAWRITER_QOS_DEFAULT;
+    dwqos.durability().kind = TRANSIENT_DURABILITY_QOS;
+    dwqos.properties().properties().emplace_back("dds.persistence.guid",
+            "77.72.69.74.65.72.5f.70.65.72.73.5f|67.75.69.64");
+    DataWriter* writer = publisher->create_datawriter(topic, dwqos);
+    ASSERT_NE(writer, nullptr);
+
+    // Configure DataReaders's durability and persistence GUID so it can use the persistence service
+    DataReaderQos drqos = DATAREADER_QOS_DEFAULT;
+    drqos.durability().kind = TRANSIENT_DURABILITY_QOS;
+    drqos.properties().properties().emplace_back("dds.persistence.guid",
+            "77.65.61.64.65.72.5f.70.65.72.73.5f|68.76.70.65");
+    DataReader* reader = subscriber->create_datareader(topic, drqos);
+    ASSERT_NE(reader, nullptr);
+
+    writer->write(&hello);
+
+    int result1 = system(
+        "python3 check_guid.py 'persistence.db' 'writers' '77.72.69.74.65.72.5f.70.65.72.73.5f|67.75.69.64'");
+    ASSERT_EQ((result1 >> 8), 1);
+
+    int result2 = system(
+        "python3 check_guid.py 'persistence.db' 'readers' '77.65.61.64.65.72.5f.70.65.72.73.5f|68.76.70.65'");
+    ASSERT_EQ((result2 >> 8), 1);
+
+    std::remove("persistence.db");
```

Comment:
These kind of thing are suitable candidates for `TearDown()` (and maybe `SetUp()`)